pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -480,31 +480,40 @@ class Build {
         }
         def sdkUrl = "${env.BUILD_URL}/artifact/workspace/target/${filter}/*zip*/target.zip"
         context.echo "sdkUrl is ${sdkUrl}"
-        def targets = ['sanity.jck', 'extended.jck', 'special.jck']
-        def parallel = 'None'
-        def num_machines = '1'
         def remoteTargets = [:]
         def additionalTestLabel = buildConfig.ADDITIONAL_TEST_LABEL
 
-        targets.each { target ->
-            // For each requested test, i.e 'sanity.jck', 'extended.jck', 'special.jck', call test job
+        def targets = ['serial': 'sanity.jck,extended.jck,special.jck']
+
+        if ("${platform}" == 'x86-64_linux' || "${platform}" == 'x86-64_windows' || "${platform}" == 'x86-64_mac') {
+            // Primary platforms run extended.jck in Parallel
+            targets['serial']   = 'sanity.jck,special.jck'
+            targets['parallel'] = 'extended.jck'
+        }
+
+        targets.each { targetKey, targetValue -> 
             try {
-                context.println "Remote trigger ${target}"
-                remoteTargets["${target}"] = {
-                    if ( "${target}" == 'extended.jck' && ("${platform}" == 'x86-64_linux' || "${platform}" == 'x86-64_windows' || "${platform}" == 'x86-64_mac') ) {
-                        parallel = 'Dynamic'
-                        num_machines = '2'
+                context.println "Remote trigger: ${targetValue}"
+                remoteTargets["${targetValue}"] = {
+                    def displayName = "${buildConfig.SCM_REF} : ${platform} : ${targetValue}"
+                    def parallel = 'None'
+                    def num_machines = '1'
+                    if ("${targetKey}" == 'parallel') {
+                         parallel = 'Dynamic'
+                         num_machines = '2'
```

Comment:
Can we go higher?  Should it be a passed in value?

Reply:
@karianna  I think the parallel or not is an aspect of the remote test triggering, the caller is simply requesting the "platform" to trigger on...
unless you meant something different?

Reply:
@karianna 2 seems to be optimal for meeting our targets with the systems we have. Happy to discuss this offline if you wish.